api_tests/lib/bitcoind_instance.ts

```diff
@@ -18,39 +18,25 @@ export class BitcoindInstance implements LedgerInstance {
         private readonly logDir: string,
         public readonly p2pPort: number,
         public readonly rpcPort: number
-    ) {}
+    ) {
+        this.dbDir = tmp.dirSync();
```

Comment:
We are mixing sync and async code here. If you need an async constructor, you can use the factory method pattern and have a static async fn on this class.